akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedEventAdapterSpec.scala

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.scaladsl
+
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.testkit.typed.scaladsl.TestProbe
+import akka.actor.typed.ActorRef
+import akka.actor.typed.scaladsl.Behaviors
+import akka.persistence.query.EventEnvelope
+import akka.persistence.query.PersistenceQuery
+import akka.persistence.query.Sequence
+import akka.persistence.query.journal.leveldb.scaladsl.LeveldbReadJournal
+import akka.persistence.typed.EventAdapter
+import akka.persistence.typed.EventSeq
+import akka.persistence.typed.PersistenceId
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import akka.testkit.EventFilter
+import akka.testkit.TestEvent.Mute
+import com.typesafe.config.ConfigFactory
+import org.scalatest.WordSpecLike
+
+object EventSourcedEventAdapterSpec {
+
+  private val conf = ConfigFactory.parseString(s"""
+      akka.loggers = [akka.testkit.TestEventListener]
+      akka.persistence.journal.leveldb.dir = "target/typed-persistence-${UUID.randomUUID().toString}"
+      akka.persistence.journal.plugin = "akka.persistence.journal.leveldb"
+    """)
+
+  case class Wrapper(t: String)
+  class WrapperEventAdapter extends EventAdapter[String, Wrapper] {
+    override def toJournal(e: String): Wrapper = Wrapper("<" + e)
+    override def fromJournal(p: Wrapper, manifest: String): EventSeq[String] = EventSeq.single(p.t + ">")
+    override def manifest(event: String): String = ""
+  }
+
+  class FilterEventAdapter extends EventAdapter[String, String] {
+    override def toJournal(e: String): String = e.toUpperCase()
+
+    override def fromJournal(p: String, manifest: String): EventSeq[String] = {
+      if (p == "B") EventSeq.empty
+      else EventSeq.single(p)
+    }
+
+    override def manifest(event: String): String = ""
+  }
+
+  class SplitEventAdapter extends EventAdapter[String, String] {
+    override def toJournal(e: String): String = e.toUpperCase()
+
+    override def fromJournal(p: String, manifest: String): EventSeq[String] = {
+      EventSeq(p.map("<" + _.toString + ">"))
+    }
+
+    override def manifest(event: String): String = ""
+  }
+
+  class EventAdapterWithManifest extends EventAdapter[String, String] {
+    override def toJournal(e: String): String = e.toUpperCase()
+
+    override def fromJournal(p: String, manifest: String): EventSeq[String] = {
+      EventSeq.single(p + manifest)
```

Comment:
Heh, that's a nice hack to be able to verify the manifest was retained... would be nicer to have a test where the manifest is used for something 'real', though.

Reply:
I added the `manifest` because it is supported in Classic and backend journals support it. It has the same purpose as string manifest in serializers. Leaving it as this here because I don't think it's an important feature.